### PR TITLE
MTM-50155: Updated example microservices for java

### PR DESCRIPTION
**Pull request by pawelkowalczyksag**

### CODE REVIEW

File: content/microservice-sdk/java-bundle/java-microservice.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ Server: Docker Engine - Community
 You can download the source code of this example from our [GitHub](https://github.com/SoftwareAG/cumulocity-examples/tree/develop/hello-world-microservice) repository to build and run it using your favorite IDE, or follow the instructions below to guide you step-by-step for you to have a better understanding of the code and what needs to be done/configured.
 
 {{< c8y-admon-important >}}
-This microservice example has been tested under macOS, Ubuntu 18 and Windows 10 with Java 13, Maven 3.6.0, Docker 20.10.14; Eclipse 2019.03 and IntelliJ IDEA 2019.2 as IDE. Other tools or Java versions may require different configurations.
+This microservice example has been tested under macOS, Ubuntu 20 and Windows 10 with Java 17, Maven 3.8.5, Docker 20.10.14; Eclipse 2019.03 and IntelliJ IDEA 2022.3.2 as IDE. Other tools or Java versions may require different configurations.
```
Comment (jkot-softwareag) [inline content/microservice-sdk/java-bundle/java-microservice.md:67]: I would remove information about `Eclipse 2019.03` if we dont test it in this IDE

File: content/microservice-sdk/java-bundle/java-microservice.md [hunk 2]
```diff
@@ -331,12 +331,18 @@ $ echo -n t0071234/testuser:secret123 | base64
...
 ```
 
-and your authorization header would look like `"Authorization": "Basic dDAwNzEyMzQvdGVzdHVzZXI6c2VjcmV0MTIz"`. Employing the cURL command you can test your microservice as follows:
+and your authorization header would look like `Authorization: Basic dDAwNzEyMzQvdGVzdHVzZXI6c2VjcmV0MTIz`. Employing the cURL command you can test your microservice as follows:
 
 ```shell
 $ curl -H "Authorization: <AUTHORIZATION>" https://<yourTenantDomain>/service/hello-microservice-java/hello?name=Skywalker
 ```
 
+Most tools should already have support for Cumulocity Authorization header out of box. Simply use `<ttenantId>/<tusername>` as username and `<password>` as password. For example in modern versions, the above cURL command can also look like below, and the header will be generated for us:
```
Comment (jkot-softwareag) [inline content/microservice-sdk/java-bundle/java-microservice.md:340]: why `t`tenantId and `t`username ?

Suggestion:
`and the header will be generated for us:` ===> `and the header will be generated automatically:`

File: content/microservice-sdk/java-bundle/microservice-java.md [hunk 3]
```diff
@@ -374,7 +374,9 @@ public class App {
     }
 
     // Track client's approximate location
-    @RequestMapping("location/track")
+    @RequestMapping(
+                    value = "location/track",
+                    produces="application/json")
```
Comment (kaczynskid-sag) [inline content/microservice-sdk/java-bundle/microservice-java.md:379]: trivial:
```suggestion
    @RequestMapping(value = "location/track", produces="application/json")
```

File: content/microservice-sdk/java-bundle/troubleshooting.md [hunk 4]
```diff
@@ -81,7 +81,7 @@ This means that the process 12985 is using the 8080 port and it can be killed if
 
 ##### When trying to build an application I get a "BeanCreationException: Error creating bean with name methodSecurityInterceptor" error
 
-This is caused mainly by versions incompatibility between the SDK and Spring Boot specified in your _pom.xml_ file. If you want to use a recent version of the SDK, for example, 10.4.0, the version of Spring Boot must be 1.5.17 or later.
+This is caused mainly by versions incompatibility between the SDK and Spring Boot specified in your _pom.xml_ file. If you want to use a recent version of the SDK, for example, 1016.0.0, the version of Spring Boot must be 2.x.y.
```
Comment (kaczynskid-sag) [inline content/microservice-sdk/java-bundle/troubleshooting.md:84]:
```suggestion
This is caused mainly by versions incompatibility between the SDK and Spring Boot specified in your _pom.xml_ file. If you want to use a recent version of the SDK, for example, 1016.0.0, the version of Spring Boot must be compatible or equal to version 2.5.4.
```

File: content/microservice-sdk/java-bundle/microservice-java.md [hunk 5]
```diff
@@ -131,8 +131,8 @@ BODY:
   {
     "c8y_IsDevice": {},
     "c8y_Position": {
-      "lat": <LATITUDE>,
-      "lng": <LONGITUDE>
+      "lat": "<LATITUDE>",
+      "lng": "<LONGITUDE>"
```
Comment (jkot-softwareag) [inline content/microservice-sdk/java-bundle/microservice-java.md:135]: I think previous was proper, LATITUDE and LONGITUDE are numeric values, not strings. Please double check this.

File: content/microservice-sdk/java-bundle/developing-microservice.md [hunk 6]
```diff
@@ -297,7 +297,7 @@ The package module provides a Maven plugin to prepare a ZIP file required by the
 <plugin>
     <groupId>com.nsn.cumulocity.clients-java</groupId>
     <artifactId>microservice-package-maven-plugin</artifactId>
-    <version>9.16.2</version>
+    <version>${c8y.version}</version>
```
Comment (karolkaczmarski-sag) [inline content/microservice-sdk/java-bundle/developing-microservice.md:300]: is it defined anywhere?

Comment (pawelkowalczyksag) [reply content/microservice-sdk/java-bundle/developing-microservice.md:300]: In previoues examples yes, both times. Though at first its not there from the beginning hm... maybe it should be consolidated properly in one place? Spreaded knowledge confuse users usually

File: content/microservice-sdk/java-bundle/java-microservice.md [hunk 7]
```diff
@@ -440,7 +446,13 @@ The response looks like this:
 
 #### Run the Docker container
 
-The Docker image was built and added to the local Docker repository during the [Maven build](#build-the-microservice-application). You can list all the Docker images available with the following command:
+The Docker image was built using your local Docker repository during the [Maven build](#build-the-microservice-application). **Please note, that by default image is deleted to keep your registry clean during development**. You can change this by adding property `microservice.package.deleteImage=false` to the maven command or pom.xml.
```
Comment (BeateRixen) [inline content/microservice-sdk/java-bundle/java-microservice.md:449]:
```suggestion
The Docker image was built using your local Docker repository during the [Maven build](#build-the-microservice-application). **Note, that by default the image is deleted to keep your registry clean during development**. You can change this by adding the property `microservice.package.deleteImage=false` to the maven command or *pom.xml*.
```

File: content/microservice-sdk/java-bundle/microservice-java.md [hunk 8]
```diff
@@ -447,7 +447,7 @@ Refer to [Application library](/web/libraries/#application) in the *Web SDK guid
 
 #### Run the Docker container
 
-The Docker image is built and added to the local Docker repository during the [Maven build](#build-the-microservice-application).
+The Docker image is built and added to the local Docker repository during the [Maven build](#build-the-microservice-application) when following property is set `microservice.package.deleteImage=false`.
```
Comment (BeateRixen) [inline content/microservice-sdk/java-bundle/microservice-java.md:450]:
```suggestion
The Docker image is built and added to the local Docker repository during the [Maven build](#build-the-microservice-application) if the following property is set `microservice.package.deleteImage=false`.
```

**Review by karolkaczmarski-sag — APPROVED** [review]

**Review by kaczynskid-sag — APPROVED** [review]

File: content/microservice-sdk/java-bundle/java-microservice.md [hunk 9]
```diff
@@ -331,12 +331,18 @@ $ echo -n t0071234/testuser:secret123 | base64
...
 ```
 
-and your authorization header would look like `"Authorization": "Basic dDAwNzEyMzQvdGVzdHVzZXI6c2VjcmV0MTIz"`. Employing the cURL command you can test your microservice as follows:
+and your authorization header would look like `Authorization: Basic dDAwNzEyMzQvdGVzdHVzZXI6c2VjcmV0MTIz`. Employing the cURL command you can test your microservice as follows:
 
 ```shell
 $ curl -H "Authorization: <AUTHORIZATION>" https://<yourTenantDomain>/service/hello-microservice-java/hello?name=Skywalker
 ```
 
+Most tools should already support the Cumulocity IoT Authorization header out of the box. Simply use `<tenantId>/<username>` as username and `<password>` as password. In modern versions, for example, the above cURL command can also look like below, and the header will be generated automatically:
```
Comment (BeateRixen) [inline content/microservice-sdk/java-bundle/java-microservice.md:340]:
```suggestion
Most tools should already support the {{< product-c8y-iot >}} Authorization header out of the box. Simply use `<tenantId>/<username>` as username and `<password>` as password. In modern versions, for example, the above cURL command can also look like below, and the header will be generated automatically:
```

**Comment by jkot-softwareag** [comment]
@pawelkowalczyksag , @BeateRixen , @rssag - I recommend to graft those changes back to 10.16 branch (and to 10.17 if it is missing). Any of you have something against this?

Changes are covering a lot of version updates, and examples are using versions 10.16.0.X tested by Paweł, but we do not have it on our webpage...

**Comment by rssag** [comment]
it should then also be applied to 10.18.0

we can use the auto-backport function

**Comment by jkot-softwareag** [comment]
> it should then also be applied to 10.18.0
> 
> we can use the auto-backport function

Maybe it was merged to develop between 10.17 and 10.18, so that is why auto-backport is not working.

**Comment by rssag** [comment]
yes, cherry-pick was empty, and this seems to have been merged a while ago

**Comment by rssag** [comment]
10.16.0 is the current version. it'll be deployed later today, then the change should show up on the website

**Pull request merged**